src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -197,8 +197,11 @@ public Task Stop()
 
                             await ProduceEnd();
 
-                            // Finish reading the request body in case the app did not.
-                            await RequestBody.CopyToAsync(Stream.Null);
+                            if (!MessageBody.LocalIntakeFin)
```

Comment:
This could be `!MessageBody.LocalIntakeFin && _keepAlive`. If it's not a keep-alive request there is no risk of corrupting the next request.

Is this better?